lambdas/service/app.py

```diff
@@ -498,21 +498,25 @@ def openapi():
     '''),
 }
 
-fast_health_keys = {
-    **{
-        prop.key: format_description(prop.description)
-        for prop in HealthController.fast_properties['service']
-    },
-    **health_up_key
-}
+fast_health_keys = dict(sorted(
```

Comment:
Sorry for waffling on this but I think I prefer using tuples:

```diff
Index: lambdas/service/app.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/lambdas/service/app.py b/lambdas/service/app.py
--- a/lambdas/service/app.py	(revision 445cd50de9a6b9b3893b0f77583ab7fc7eeae30a)
+++ b/lambdas/service/app.py	(date 1624309970166)
@@ -498,25 +498,21 @@
     '''),
 }
 
-fast_health_keys = dict(sorted(
-    {
-        **{
-            prop.key: format_description(prop.description)
-            for prop in HealthController.fast_properties['service']
-        },
-        **health_up_key
-    }.items()
-))
+fast_health_keys = {
+    **{
+        prop.key: format_description(prop.description)
+        for prop in HealthController.fast_properties['service']
+    },
+    **health_up_key
+}
 
-health_all_keys = dict(sorted(
-    {
-        **{
-            prop.key: format_description(prop.description)
-            for prop in HealthController.all_properties
-        },
-        **health_up_key
-    }.items()
-))
+health_all_keys = {
+    **{
+        prop.key: format_description(prop.description)
+        for prop in HealthController.all_properties
+    },
+    **health_up_key
+}
 
 
 def health_spec(health_keys: dict):
Index: src/azul/health.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/health.py b/src/azul/health.py
--- a/src/azul/health.py	(revision 445cd50de9a6b9b3893b0f77583ab7fc7eeae30a)
+++ b/src/azul/health.py	(date 1624311789063)
@@ -7,6 +7,7 @@
 import json
 import time
 from typing import (
+    AbstractSet,
     Iterable,
     Mapping,
     Set,
@@ -231,16 +232,16 @@
                 body = cache['health']
         return body
 
-    fast_properties: Mapping[str, Set[health_property]] = {
-        'indexer': {
+    fast_properties: Mapping[str, Iterable[health_property]] = {
+        'indexer': (
             elasticsearch,
             queues,
             progress
-        },
-        'service': {
+        ),
+        'service': (
             elasticsearch,
             api_endpoints,
-        }
+        )
     }
 
     def _as_json_fast(self) -> JSON:
@@ -251,10 +252,10 @@
         self.storage_service.put(object_key=f'health/{self.lambda_name}',
                                  data=json.dumps(health_object).encode())
 
-    all_properties: Set[health_property] = {
+    all_properties: Iterable[health_property] = {
         p for p in locals().values() if isinstance(p, health_property)
     }
 
     @classmethod
-    def all_keys(cls) -> Set[str]:
-        return {p.key for p in cls.all_properties}
+    def all_keys(cls) -> AbstractSet[str]:
+        return frozenset(p.key for p in cls.all_properties)
```

Reply:
@hannes-ucsc I think there still needs to be sorting done on the keys within `service.app.health_*_keys`...

Reply:
Why?